output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/Main.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.outputcomparator.cli;
+
+import com.beust.jcommander.JCommander;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.mobilitydata.gtfsvalidator.outputcomparator.io.ValidationReport;
+
+public class Main {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final String ACCEPTANCE_REPORT_JSON = "acceptance_report.json";
+
+  public static void main(String[] argv) {
+    Arguments args = new Arguments();
+    new JCommander(args).parse(argv);
+    File[] reportDirectory = new File(args.getReportDirectory()).listFiles();
+    if (reportDirectory == null) {
+      logger.atSevere().log("Specified output is not a directory, or an I/O error occurred");
+      return;
+    }
+    if (reportDirectory.length == 0) {
+      logger.atSevere().log(
+          "Specified directory is empty, cannot generate acceptance tests report.");
+      return;
+    }
+    ImmutableMap.Builder<String, Integer> mapBuilder = new Builder<>();
+    int badDatasetCount = 0;
+    List<File> reportDirs =
+        Arrays.stream(reportDirectory).filter(File::isDirectory).collect(Collectors.toList());
+
+    int totalDatasetCount = reportDirs.size();
+
+    try {
+      for (File file : reportDirs) {
+        int newErrorCount = 0;
+        ValidationReport referenceReport =
+            ValidationReport.fromPath(
+                file.toPath().resolve(args.getReferenceValidationReportName()));
+        ValidationReport latestReport =
+            ValidationReport.fromPath(file.toPath().resolve(args.getLatestValidationReportName()));
+
+        if (referenceReport.hasSameErrorCodes(latestReport)) {
+          mapBuilder.put(file.getName(), newErrorCount);
+          continue;
+        }
+        if (referenceReport.equals(latestReport)) {
```

Comment:
> If the two ValidationReports have different error codes (hasSameErrorCodes() returns false above), then they can't be equal, right?

Indeed. Thanks for flagging that - removing it since the goal here is only to check differences in the listing of error types.